adb/filesync_protocol.py

```diff
@@ -107,11 +123,19 @@ def Push(cls, connection, datafile, filename,
     cnxn = FileSyncConnection(connection, b'<2I')
     cnxn.Send(b'SEND', fileinfo)
 
+    if progress_callback:
+      progress = cls._HandlePushProgress(filename, datafile, progress_callback)
```

Comment:
Going with a generator is okay, but you're passing a lot of unnecessary context into it. If you instead passed in a lambda like this:

```py
total_bytes = os.fstat(datafile.fileno()).st_size if isinstance(datafile, file) else -1
progress = cls._HandleProgress(lambda current: progress_callback(filename, current, total_bytes))
```

Now, `_HandleProgress` only tracks things related to the progress: the current bytes and exception ignoring. We could reuse this for anything else that needs progress tracked, like Pull